jax/interpreters/partial_eval.py

```diff
@@ -378,21 +415,21 @@ def trace_to_jaxpr(fun: lu.WrappedFun, pvals: Sequence[PartialVal],
   return jaxpr, out_pvals, consts
 
 @lu.transformation
-def trace_to_subjaxpr(master, instantiate, pvals):
+def trace_to_subjaxpr(master: core.MasterTrace, instantiate: Union[bool, Sequence[bool]],
+                      pvals: Sequence[PartialVal]):
   assert all([isinstance(pv, PartialVal) for pv in pvals]), pvals
   trace = JaxprTrace(master, core.cur_sublevel())
   in_tracers = map(trace.new_arg, pvals)
   ans = yield in_tracers, {}
-  instantiate = [instantiate] * len(ans) if type(instantiate) is bool else instantiate
+  instantiate = [instantiate] * len(ans) if isinstance(instantiate, bool) else instantiate
```

Comment:
I don't know if it applies in this case, but mypy special cases `isinstance` checks when trying to understand type tests, but does not handle `type(...) is ...`.

Reply:
Indeed, I changed it because mypy complained.

Reply:
Thanks, that's what I was wondering.

I phrased my comment really badly here. Sorry. I meant to be asking a question. Thanks for explaining, and teaching me something!

Reply:
I did some simple %timeit. If x is a tuple of booleans then 
                                             x is bool         x is tuple or list of bools
type(x) is bool                    90ns              104ns
isinstance(x, bool)             80ns              104ns

Here is a more interesting one: if we build tuples of booleans instead of lists of booleans
In [12]: %timeit [True] * 5                                                     
100 ns ± 2.5 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [13]: %timeit (True,) * 5                                                    
7.52 ns ± 0.0874 ns per loop (mean ± std. dev. of 7 runs, 100000000 loops each)

It seems that we could save more by using tuples instead of lists, unless we need mutation, which is a rare exception in JAX.